packages/core/src/actions/smartAccount/buildUserOperation.ts

```diff
@@ -60,15 +53,21 @@ export async function buildUserOperation<
   const _uo =
     entryPoint.version === "0.6.0"
       ? ({
-          initCode: getInitCode,
+          initCode: account.getInitCode(),
           sender: account.address,
           nonce,
           callData,
           signature,
         } as Deferrable<UserOperationStruct<TEntryPointVersion>>)
       : ({
-          factory: getFactoryAndData.then(([factory]) => factory),
-          factoryData: getFactoryAndData.then(([, data]) => data),
+          ...account.isAccountDeployed().then((deployed) =>
+            deployed
+              ? {}
+              : {
+                  factory: account.getFactoryAddress(),
+                  factoryData: account.getFactoryData(),
+                }
+          ),
```

Comment:
Does this work? This looks like it's trying to spread a promise which won't work because the promise doesn't have spreadable properties that make sense to add on to the uo struct.

Reply:
good catch. no it doesn't work as I tried. Gotta await and resolve first for spread.